src/protocol-handler-installer.js

```diff
@@ -0,0 +1,49 @@
+const {CompositeDisposable} = require('event-kit')
+
+const {remote} = require('electron')
+
+module.exports =
+class ProtocolHandlerInstaller {
+  constructor () {
+    this.subscriptions = new CompositeDisposable()
+    this.supported = ['win32', 'darwin'].includes(process.platform)
+  }
+
+  initialize (config) {
+    this.config = config
+
+    this.subscriptions.add(
+      this.config.observe('core.defaultProtocolHandler', this.onValueChange.bind(this))
```

Comment:
🤔  I wonder if it's too obnoxious to immediately prompt the user if they switch to `"prompt"` and Atom isn't the default protocol handler — maybe we should only check at startup?